src/main/java/com/usharesoft/jenkins/steps/InstallStep.java

```diff
@@ -18,22 +18,43 @@ public InstallStep(UForgeLauncher launcher, String version) {
     @Override
     public void perform() throws InterruptedException, IOException {
         printStep(Messages.Logs_steps_install());
-        launcher.launchInstall(getInstallVenvCmd(), true);
+        launcher.launchInstall(getDownloadVenvCmd(), true);
+        launcher.launchInstall(getExtractVenvCmd(), true);
+        launcher.launchInstall(getInitVenvCmd(), true);
         launcher.launchInstall(getInstallHammrCmd(), true);
     }
 
-    ArgumentListBuilder getInstallVenvCmd() {
+    ArgumentListBuilder getDownloadVenvCmd() {
         ArgumentListBuilder args = new ArgumentListBuilder();
-        args.add("virtualenv");
-        args.add("--python=python2.7");
-        args.add(launcher.getScriptWorkspace());
+        args.add("curl");
+        args.add("--location");
+        args.add("--output").add("virtualenv.tar.gz");
+        args.add("https://github.com/pypa/virtualenv/tarball/12.0.7");
```

Comment:
If we decide to fix the venv version here, why we don't get the latest one today: 16.6.1

Reply:
True, I'll make the change